### ISSUE TITLE
Widget node depends with block parameter

### ISSUE THREAD

**Issue by gabs77** [magento/magento2#7252]
On Magento 2.1.2, When we create specific widget, the node "depends" has some difficulties to turn well.

        <parameter name="url_type" xsi:type="select" visible="true" required="true">
                <label translate="true">Type Link Bloc</label>
                <options>
                    <option name="product" value="product" selected="true">
                        <label translate="true">Product</label>
                    </option>
                    <option name="custom" value="custom">
                        <label translate="true">Custom</label>
                    </option>
                </options>
            </parameter>
            <parameter name="id_path_product" xsi:type="block" visible="true">
                <label translate="true">Product</label>
                <depends>
                    <parameter name="url_type" value="product"/>
                </depends>
                <block class="Magento\Catalog\Block\Adminhtml\Product\Widget\Chooser">
                    <data>
                        <item name="button" xsi:type="array">
                            <item name="open" xsi:type="string" translate="true">Select Product...</item>
                        </item>
                    </data>
                </block>
            </parameter>
            <parameter name="custom_url" xsi:type="text" visible="true">
                <label translate="true">Link Bloc</label>
                <depends>
                    <parameter name="url_type" value="custom"/>
                </depends>
            </parameter>

### Expected result
When the user select the type parameter as product, the custom link parameter has hidden and the product link has show.
When the user select the type parameter as custom, the product link parameter has show and the product link has hide.

### Actual result
When the user select the type parameter as product, the custom link parameter has hidden. It's perfect.
When the user select the type parameter as custom, the product link parameter has always shown.

**Comment by veloraven** [comment]
@gabs77 thank you for your feedback.
Please, format this issue according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines): with steps to reproduce, actual result and expected result.

**Comment by biolauri** [comment]
It seems to me that this is a duplicate of #6868 and faces exact the same problem. Although I think, the problem is described a bit better here.

**Comment by gabs77** [comment]
Hi

 
I suggest a first solution

```
diff --git a/lib/web/mage/adminhtml/form.js b/lib/web/mage/adminhtml/form.js
index 1d137dc..b6dcabf 100644
--- a/lib/web/mage/adminhtml/form.js
+++ b/lib/web/mage/adminhtml/form.js
@@ -359,6 +359,7 @@ FormElementDependenceController.prototype = {
     {
         // define whether the target should show up
         var shouldShowUp = true;
+        var chooserBlock = false;
         for (var idFrom in valuesFrom) {
             var from = $(idFrom);
             if (from) {
@@ -377,6 +378,11 @@ FormElementDependenceController.prototype = {
             target = $(idTo);

         // Target won't always exist (for example, if field type is "label")
+        if (!target) {
+            target = $('chooser'+idTo);
+            chooserBlock = true;
+        }
+
         if (target) {
             var inputs = target.up(this._config.levels_up).select('input', 'select', 'td'),
                 isAnInputOrSelect = ['input', 'select'].indexOf(target.tagName.toLowerCase()) != -1;
@@ -440,7 +446,11 @@ FormElementDependenceController.prototype = {
         }
         var rowElement = $('row_' + idTo);
         if (rowElement == undefined && target) {
-            rowElement = target.up(this._config.levels_up);
+            var levelUp = this._config.levels_up;
+            if (chooserBlock) {
+                levelUp = this._config.levels_up + 1;
+            }
+            rowElement = target.up(levelUp);
         }
         if (rowElement) {
             if (shouldShowUp) {

```
It's not perfect but it's hide a button chooser block

**Comment by magento-engcom-team** [comment]
Duplicate of #6868. internal ticket MAGETWO-82054 already created

**Comment by magento-engcom-team** [comment]
Hi @gabs77. Thank you for your report. 
The issue has been fixed in magento/magento2#30570 by @hostep in 2.4-develop branch
Related commit(s):
 - [e3789f0906f45b565fdb8d13d28c305917cca1bd](https://github.com/magento/magento2/commit/e3789f0906f45b565fdb8d13d28c305917cca1bd)

The fix will be available with the upcoming 2.4.2 release.
